package: Do not package apps

ddn0 opened the pull request:
Applications take a lot of space (500 MB) and they slow down packaging
because we spend time compressing, uncompressing and hashing them. No
one needs to use them currently so omit them from any packages we make.

l-hoang commented on libquery/src/SubgraphQuery.cpp:
```diff
@@ -494,15 +494,59 @@ class SubgraphQueryMiner {
+      // pick the first non-inserted vertex that is connected to an already
+      // inserted vertex only consider DIRECTED_EDGE for connection
+      for (VertexId j = 1; j < queryGraph.size(); ++j) {
+        auto vertex = vertexToMatchingOrderMap[j];
```
i would prefer auto *never* be used if the type is short enough to use, but that's just a nitpick; not required for this PR, i can get to it later :P

roshandathathri replied:
I would prefer auto whether the type does not matter. It will help us to change underlying types if and when needed.

l-hoang commented on libquery/src/SubgraphQuery.cpp:
```diff
@@ -494,15 +494,59 @@ class SubgraphQueryMiner {
+      // inserted vertex only consider DIRECTED_EDGE for connection
+      for (VertexId j = 1; j < queryGraph.size(); ++j) {
+        auto vertex = vertexToMatchingOrderMap[j];
+        if (inserted.find(vertex) == inserted.end()) {
```
some comments here explaining what each of these ifs is doing would be nice;

my understanding (correct if wrong)

first, check if candidate is outgoing neighbor of existing node
if not found, check if it connects to existing node

add if either above is true

not required for now, will get to it later

roshandathathri replied:
The comments at the beginning (before line 513) intend to do that. 

I didn't understand your explanation. "connects" can be either outgoing or incoming. 

The candidate vertex (if not already inserted) will check if it has an outgoing/incoming neighbor to an existing vertex. If it does, then it will be inserted.

l-hoang replied:
yes, ultimately both "ifs" are checking for some connection

l-hoang commented on libquery/src/SubgraphQuery.cpp:
```diff
@@ -494,15 +494,59 @@ class SubgraphQueryMiner {
+            inserted.insert(vertex);
+            break;
+          }
+        }
```
you only add something if connected is true; are you ignoring disconnected nodes (for now?)?

roshandathathri replied:
The entire subgraph is expected to be connected for now.

ddn0 committed:
build: Do not package or install apps

Applications take a lot of space (500 MB) and they slow down packaging
because we spend time compressing, uncompressing and hashing them. No
one needs to use them currently so omit them from any packages we make.

arthurp reviewed (changes requested):
I think this will cause a `-apps` to be an empty or nearly empty debian package. Lets eliminate the package too. If we need `katana-enterprise-apps` still, then the setup code for `-apps` can be moved to the enterprise build, or we can add a flag to the macro that sets up the packaging.

The package setup in question is here:

https://github.com/KatanaGraph/katana/blob/142ec9578101ffc5299f02473dc9d214100a7698/cmake/Modules/KatanaBuildCommon.cmake#L488-L493

arthurp commented:
Oh another thing, do we want to remove these apps from packaging, or would it be better, to just not install them at all? Is there ever a reason to install them? If they are just for testing, then they can be fun from the build directory.

ddn0 commented:
The current PR actually does achieve the stated effect, here's the output of `cpack` with both the enterprise and open builds.

Enterprise:
```
ddn@spork:katana-build$ cpack
CPack: Create package using DEB
CPack: Install projects
CPack: - Run preinstall target for: Katana-enterprise
CPack: - Install project: Katana-enterprise []
CPack: -   Install component: dev
CPack: -   Install component: enterprise-apps
CPack: -   Install component: enterprise-dev
CPack: -   Install component: enterprise-lib
CPack: -   Install component: enterprise-shlib
CPack: -   Install component: enterprise-tools
CPack: -   Install component: lib
CPack: -   Install component: shlib
CPack: -   Install component: tools
CPack: -   Install component: z-enterprise-shlib
CPack: Create package
CPackDeb: - Generating dependency list
CPackDeb: - Generating dependency list
CPackDeb: - Generating dependency list
CPack: - package: /home/ddn/w/katana-build/pkg/katana-enterprise-apps-static_0.1.12~dev+70.192.5d9535.d7fbc8.unmerged_amd64.deb generated.
CPack: - package: /home/ddn/w/katana-build/pkg/libkatana-enterprise-dev-static_0.1.12~dev+70.192.5d9535.d7fbc8.unmerged_amd64.deb generated.
CPack: - package: /home/ddn/w/katana-build/pkg/libkatana-enterprise-static_0.1.12~dev+70.192.5d9535.d7fbc8.unmerged_amd64.deb generated.
CPack: - package: /home/ddn/w/katana-build/pkg/katana-enterprise-tools-static_0.1.12~dev+70.192.5d9535.d7fbc8.unmerged_amd64.deb generated.
```

Open:
```
ddn@spork:katana-open$ cpack
CPack: Create package using DEB
CPack: Install projects
CPack: - Run preinstall target for: Katana
CPack: - Install project: Katana []
CPack: -   Install component: dev
CPack: -   Install component: lib
CPack: -   Install component: shlib
CPack: -   Install component: tools
CPack: Create package
CPackDeb: - Generating dependency list
CPackDeb: - Generating dependency list
CPack: - package: /home/ddn/w/katana-open/pkg/libkatana-dev-static_0.1.12~dev+70.192.5d9535.d7fbc8.unmerged_amd64.deb generated.
CPack: - package: /home/ddn/w/katana-open/pkg/libkatana-static_0.1.12~dev+70.192.5d9535.d7fbc8.unmerged_amd64.deb generated.
CPack: - package: /home/ddn/w/katana-open/pkg/katana-tools-static_0.1.12~dev+70.192.5d9535.d7fbc8.unmerged_amd64.deb generated.
```

Whether we want to immediately remove all support for the apps package in this repo is a different question. I can make that change too though having a macro defined in this repo that has a flag like "add enterprise stuff" seems weird to me, so I might rearrange things differently.

> Oh another thing, do we want to remove these apps from packaging, or would it be better, to just not install them at all? Is there ever a reason to install them? If they are just for testing, then they can be fun from the build directory.

I didn't want to make such a big change, but if people are okay with this, I can remove the `install` commands too.

arthurp commented:
> The current PR actually does achieve the stated effect,

Great. Looks like cpack doesn't generate packages which have nothing to install into them. I don't see any reason to mess with it then. Eventually we can remove the `apps` component entirely, but that's not pressing.

> if people are okay with this, I can remove the `install` commands too.

I would vote for removing install commands. But no need to do it now if it's not clear that that's the right choice. We can revisit it later. If it doesn't go in the packages it isn't a bit deal unless someone is trying to build and install from source, and I don't really think people are doing that much if at all any more.

Thanks for handling this.

ddn0 commented:
> I would vote for removing install commands. But no need to do it now if it's not clear that that's the right choice. We can revisit it later.

I just went ahead and removed the installs as well. It seems weird to install things that we don't package.

ddn0 commented:
@gurbinder533 thoughts?

gurbinder533 commented:
This looks fine to me. We don't have anyone using them as of now. Only Intel is using and testing our CPU apps and they mostly care about library API and Conda packages.
